### PR TITLE
[Woptim] Shared ci 2023.12.0

### PR DESCRIPTION
**Pull request by adayton1**
Brings in changes from #240 and adds updates for 2023.12.0 as well as changes made for CI in other repos such as CHAI.

### CODE REVIEW

**Comment by adayton1** [comment]
Supersedes https://github.com/LLNL/CARE/pull/240

**Review by liu15 — APPROVED** [review]

**Review by dtaller — APPROVED** [review]

File: .gitlab/jobs/poodle.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+###############################################################################
+# Copyright (c) 2022-24, Lawrence Livermore National Security, LLC and RADIUSS
+# project contributors. See the CHAI LICENSE file for details.
```
Comment (adrienbernede) [inline .gitlab/jobs/poodle.yml:3]: Looks like "CHAI LICENSE" is a leftover from a copy paste.

File: .gitlab/jobs/corona.yml [hunk 2]
```diff
@@ -1,16 +1,28 @@
 ##############################################################################
-# Copyright (c) 2022, Lawrence Livermore National Security, LLC and RADIUSS
-# project contributors. See the COPYRIGHT file for details.
+# Copyright (c) 2022-24, Lawrence Livermore National Security, LLC and RADIUSS
+# project contributors. See the CHAI LICENSE file for details.
```
Comment (adrienbernede) [inline .gitlab/jobs/corona.yml:3]: Looks like "CHAI LICENSE" is a leftover from a copy paste.

File: .uberenv_config.json [hunk 3]
```diff
@@ -4,7 +4,7 @@
 "package_final_phase": "initconfig",
 "package_source_dir": "../..",
 "spack_url": "https://github.com/spack/spack.git",
-"spack_branch": "e4s-23.02",
+"spack_branch": "v0.20.1",
```
Comment (adrienbernede) [inline .uberenv_config.json:7]: We are now using "develop-2024-01-21".

**Review by adrienbernede — COMMENTED** [review]
@adayton1 I greatly appreciate your work here. Thanks.

File: .uberenv_config.json (hunk 3 above)
Comment (adayton1) [reply .uberenv_config.json:7]: Fixed in https://github.com/LLNL/CARE/pull/246

**Pull request merged**